src/undate/undate.py

```
@@ -81,72 +120,77 @@ def __init__(
+        )
 
-        self.label = label
+    def set_calendar(self, calendar: Union[str, Calendar]):
+        if calendar is not None:
+            # if not passed as a Calendar instance, do a lookup
+            if not isinstance(calendar, Calendar):
+                # look for calendar by upper-case name
+                try:
+                    calendar = Calendar[calendar.upper()]
+                except KeyError:
+                    raise ValueError(f"Calendar `{calendar}` is not supported")
```

Ruff (0.8.0)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)